### PR TITLE
Add pod resources and eviction to weave-kube doc

### PR DESCRIPTION
**Pull request by bboreham**
Explaining the issue described at #2794

### CODE REVIEW

**Comment by bboreham** [comment]
@abuehrle please let us know if you able to look at this, or if we should find someone else.

File: site/kube-addon.md [hunk 1]
```diff
@@ -91,6 +94,56 @@ and so you will need to perform the procedure manually:
...
+You can reduce the chance of eviction by changing the DaemonSet to
+have a much bigger request, and a limit of the same value.
+
+This causes Kubernetes to apply a ["guaranteed" rather than a
+"burstable" policy](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-qos.md).
+However a similar request for disk space can not
+be made, and so please be aware of this issue and monitor your
+resources to ensure that they stay below 100%.
+
+Kubernetes displays a notification in the event of an eviction similar to this:
```
Comment (marccarre) [inline site/kube-addon.md]: Beginners typically would not know where to look for this notification. 
Could it be worth adding an example of command to get this?

Comment (bboreham) [inline site/kube-addon.md]: Updated.

**Comment by bboreham** [comment]
Note for posterity: it took me an incredibly long time to recreate an eviction on demand; what worked best was to set a hard eviction policy as [explained in the docs](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/kubelet-eviction.md)

I added `--eviction-hard=memory.available<1000Mi` to the kubelet arguments, and then ran multiple copies of this pod:

    kubectl create -f https://raw.githubusercontent.com/derekwaynecarr/kubernetes/examples-eviction/demo/kubelet-eviction/besteffort-pod.yaml

**Pull request merged**